draft-jhoyla-req-mtls-flag.md

```diff
@@ -133,7 +133,56 @@ TLS namespace with the following values:
 
 --- back
 
+# Appendix
+
+## Alternative Approaches
+
+### Concealed Auth
+
+Concealed Auth {{?RFC9729}} is a draft that allows a client to attach a header
+at the HTTP layer that cryptographically binds a public key to the underlying
+TLS connection. This gives us a similar authentication property to using
+`request_client_auth`, and is also client initiated. This solution is applicable
+in some contexts, but in the case of authenticating bots is difficult to
+implement.
+
+The way many HTTP libraries work is that they dispatch requests to a connection
+pool. Once the request arrives at the pool it is sent on a chosen connection
+driven by internal logic. This, unfortunately, makes it difficult to know which
+connection a request will be sent on, and thus to produce the appropriate
+signature.
+
+It is possible to avoid this issue, for example by not using connection pooling,
+or by sending a special "pre-request" on each connection that's added to a pool,
+however these solutions tend to be bespoke per library and bring a number of
+their own issues (e.g. handling servers that terminate connections after a
+single request, or handling resumption).
+
+### Post-handshake authentication (PHA)
+
+Post-handshake authentication is part of the core TLS 1.3 RFC, although it is
+not widely supported. Its use incurs an extra round-trip, and would need
+negotiation at the application layer. This just trades one negotiation for
+another, whilst making the protocol more expensive to run. Further there are
+some reservations about its security guarantees.
+
+### ALPN
+
+One proposal is for the client to send new ALPN extensions, e.g. `http2+bot`.
+This is functionally equivalent to this draft, but with a less efficient
+encoding.
```

Comment:
Say why this is functionally equivalent. We might need to flesh out the proposal a bit more.